File: Brece-cs50-2018-x-vigenere/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    //only execute with ONE command line argument
    if (argc != 2)
    {
        printf("ERROR: exactly 1 argument!\n");
        return 1;
    }

    //check if key word consist on letters ONLY, print ERROR and return 1
    string key = argv[1];

    for (int i = 0, n = strlen(key); i < n; i++)
    {
        if (!isalpha(key[i]))
        {
            printf("ERROR: argument must consist on LETTERS only!\n");
            return 1;
        }
    }

    //user input
    string s = get_string("paintext: ");

    printf("ciphertext: ");

    //m increases the index j by 1 IF the char in plaintext is alpabetical
    int m = 0;
    
    //for each character in plaintext
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        //rotate index position in string key 
        int j = m % strlen(key);
        
        //transform letter in key to a number between 0 and 25
        int k = (tolower(key[j]) - 97);
        
        //if alphabetic
        if (isalpha(s[i]))
        {
            if (islower(s[i]))
            {
                printf("%c", (((s[i] - 97)  + k) % 26) + 97);
            }
            else
            {
                printf("%c", (((s[i] - 65) + k) % 26) + 65);
            }
            //next letter in key --> index j increases due to modulos by 1
            m++;
        }
        //not alphabetic
        else
        {
            printf("%c", (s[i]));
            
        }
    }





    printf("\n");
    return 0;

}

File: Brece-cs50-2018-x-sentimental-vigenere/vigenere.py
import sys
import cs50


# check for proper usage and key argument value
if len(sys.argv) != 2:
    print("Usage: python <program> <key>")
    sys.exit(1)

key = str(sys.argv[1])

if key.isalpha() == False:
    print("<key> argument must consist on letters only")
    sys.exit(1)


# get user input
print("plaintext: ", end="")
text = cs50.get_string()

print("ciphertext: ", end="")


# index for key-letter increases by one if the the char in plaintext is alphabetic
m = 0

# loop for each char in plaintext
for c in text:

    # index in string <key>
    j = m % len(key)

    # convert char in <key> to ascii number
    k = ord(key[j].lower()) - 97

    # if char in plaintext is alphabetic
    if c.isalpha():

        if c.islower():
            print(chr((((ord(c) - 97) + k) % 26) + 97), end="")

        else:
            print(chr((((ord(c) - 65) + k) % 26) + 65), end="")

        m += 1

    else:
        print(c, end="")
        
print("")